apps/daimo-mobile/src/view/screen/DeviceScreen.tsx

```diff
@@ -0,0 +1,160 @@
+
+  const deviceName = "Device " + pubKeyToEmoji(device.key);
+
+  useEffect(() => {
+    navigation.setOptions({ title: deviceName });
+  }, [navigation, deviceName]);
+
+  const nonce = useMemo(
+    () => new DaimoNonce(new DaimoNonceMetadata(DaimoNonceType.RemoveKey)),
+    [devicePubkey]
+  );
+
+  const sendFn = async (account: DaimoAccount) => {
+    console.log(`[ACTION] removing device ${devicePubkey}`);
+    return account.removeSigningKey(devicePubkey, nonce);
+  };
+
+  const { status, message, cost, exec } = useSendAsync({
+    enclaveKeyName: account.enclaveKeyName,
+    dollarsToSend: 0,
+    sendFn,
+  });
+
+  const removeDevice = useCallback(() => {
+    const { enclaveKeyName, enclavePubKey } = account;
+
+    Alert.alert(
+      "Remove " + deviceName + "\n",
+      `Are you sure you want to remove this device?\n\n` +
+        `If this is the only device on your account, you'll lose your account.`,
+      [
+        {
+          text: "Remove Device",
+          onPress: removeKey,
+        },
+        {
+          text: "Cancel",
+          style: "cancel",
+        },
+      ]
+    );
+
+    async function removeKey() {
+      console.log(`[DEVICE] Removing device ${devicePubkey}`);
+      exec();
+
+      if (devicePubkey === enclavePubKey) {
+        console.log(`[USER] deleting account; deleting key ${enclaveKeyName}`);
+        await deleteEnclaveKey(enclaveKeyName);
+
+        setAccount(null);
+        nav.navigate("Home");
+      }
+    }
+  }, []);
+
+  const statusMessage = (function (): ReactNode {
+    switch (status) {
+      case "idle":
+        return `Remove fee: ${getAmountText({ dollars: cost.totalDollars })}`;
+      case "loading":
+        return message;
+      case "error":
+        return <TextError>{message}</TextError>;
+      default:
+        return null;
+    }
+  })();
+
+  const button = (function () {
+    switch (status) {
+      case "idle":
+        return (
+          <ButtonBig
+            type="danger"
+            title="Remove Device"
+            onPress={removeDevice}
+          />
+        );
+      case "loading":
+        return <ActivityIndicator size="large" />;
+      case "success":
+        return <ButtonBig type="success" title="Success" disabled />;
+      case "error":
+        return <ButtonBig type="danger" title="Error" disabled />;
+    }
+  })();
+
+  return (
+    <View style={ss.container.vertModal}>
+      <Spacer h={16} />
+      <TextH1>{deviceName}</TextH1>
+      <Spacer h={16} />
+      {devicePubkey === account.enclavePubKey && (
+        <>
+          <TextCenter>
+            <TextH2>Current Device</TextH2>
+          </TextCenter>
+          <Spacer h={16} />
+        </>
+      )}
+      <TextBody>
+        <TextCenter>
+          Added at {timeString(guessTimestampFromNum(device.addedAt))}
+        </TextCenter>
+      </TextBody>
+      <Spacer h={16} />
+      <TextCenter>Last used at unknown</TextCenter>
```

Comment:
intentional?

Reply:
yeah stub to show we'll add it future but don't feel strongly about keeping it, also changed it to `Last used at ??? (TODO)` so it doesnt look like a bug.

Reply:
can also comment for now `{/* <TextCenter>Last used at unknown</TextCenter> */}`

Reply:
done